frontend/packages/knative-plugin/src/utils/get-knative-resources.ts

```diff
@@ -438,31 +435,87 @@ export const getSinkableResources = (namespace: string): FirehoseResource[] => {
     : [];
 };
 
-export const getKnativeServingResources = (namespace: string) => {
-  return {
-    ...knativeServingResourcesRevisionWatchers(namespace),
-    ...knativeServingResourcesConfigurationsWatchers(namespace),
-    ...knativeServingResourcesRoutesWatchers(namespace),
-    ...knativeServingResourcesServicesWatchers(namespace),
-    ...knativeCamelDomainMappingResourceWatchers(namespace),
-  };
-};
+export const getKnativeServingResources = () =>
+  Promise.resolve({
+    revisions: {
+      model: { group: 'serving.knative.dev', version: 'v1', kind: 'Revision' },
+      opts: { isList: true, optional: true, namespaced: true },
+    },
+    configurations: {
+      model: { group: 'serving.knative.dev', version: 'v1', kind: 'Configuration' },
+      opts: { isList: true, optional: true, namespaced: true },
+    },
+    ksroutes: {
+      model: { group: 'serving.knative.dev', version: 'v1', kind: 'Route' },
+      opts: { isList: true, optional: true, namespaced: true },
+    },
+    ksservices: {
+      model: { group: 'serving.knative.dev', version: 'v1', kind: 'Service' },
+      opts: { isList: true, optional: true, namespaced: true },
+    },
+    domainmappings: {
+      model: { group: 'serving.knative.dev', version: 'v1beta1', kind: 'DomainMapping' },
+      opts: { isList: true, optional: true, namespaced: true },
+    },
+  });
```

Comment:
let's do some type checking here

```suggestion
  } satisfies WatchK8sResourcesGeneric);
```